web-client/public/js/upload.js

```diff
@@ -301,30 +329,104 @@ export const upload = function (sliderObject, sliderGroupController, drawGraph,
         delay: { show: 700, hide: 100 }
     });
 
-    var grayThreshold = false;
+    // Normalization Controller
+    var MIN_EDGE_WEIGHT_NORMALIZATION = 0.0001;
+    var MAX_EDGE_WEIGHT_NORMALIZATION = 1000;
+
+    var edgeWeightNormalizationInputValidation = function (value) {
+        if (value === "") { // sets to default value
+            return "";
+        } else if (value < MIN_EDGE_WEIGHT_NORMALIZATION) {
+            return MIN_EDGE_WEIGHT_NORMALIZATION;
+        } else if (value > MAX_EDGE_WEIGHT_NORMALIZATION) {
+            return MAX_EDGE_WEIGHT_NORMALIZATION;
+        } else {
+            return value;
+        }
+    };
+    var synchronizeNormalizationValues = function (value) {
+        var validated = edgeWeightNormalizationInputValidation(value);
+        $("#normalization-max").val(validated);
+        $("#edge-weight-normalization-factor-menu").val(validated);
+    };
 
     $("#normalization-button").click(function () {
-        normalization = true;
-    // displayNetwork(currentNetwork, name, normalization);
-        drawGraph(currentNetwork, sliders, normalization, grayThreshold, nodeColoring);
+        synchronizeNormalizationValues($("#normalization-max").val());
+        drawGraph(currentNetwork, sliders, nodeColoring);
     });
 
-    $("#resetNormalizationButton").click(function () {
-        document.getElementById("normalization-max").value = "";
-    // normalization = false;
-    // displayNetwork(currentNetwork, name, normalization);
-        drawGraph(currentNetwork, sliders, normalization, grayThreshold, nodeColoring);
+    $("#reset-normalization-factor-menu, #resetNormalizationButton").click(function () {
+        synchronizeNormalizationValues("");
+        drawGraph(currentNetwork, sliders, nodeColoring);
     });
 
-    $("#grayThresholdInput").on("change", function () {
-        grayThreshold = true;
-    // displayNetwork(currentNetwork, name, normalization, grayThreshold);
-        drawGraph(currentNetwork, sliders, normalization, grayThreshold, nodeColoring);
+    $("#edge-weight-normalization-factor-menu").on("change", function () {
+        synchronizeNormalizationValues($("#edge-weight-normalization-factor-menu").val());
+        drawGraph(currentNetwork, sliders, nodeColoring);
```

Comment:
It looks like `synchronizeNormalizationValues` is always followed by `drawGraph`—so can `drawGraph` be the last line of `synchronizeNormalizationValues` instead? Then the event handlers will be one-liners.

Reply:
Done!